server/graphql/v2/mutation/LegalDocumentsMutations.ts

```diff
@@ -0,0 +1,124 @@
+import assert from 'assert';
+
+import debugLib from 'debug';
+import { GraphQLNonNull } from 'graphql';
+import { GraphQLJSON } from 'graphql-scalars';
+import { encodeBase64 } from 'tweetnacl-util';
+
+import ActivityTypes from '../../../constants/activities';
+import { notify } from '../../../lib/notifications/email';
+import { getUSTaxFormPdf } from '../../../lib/pdf';
+import { reportErrorToSentry } from '../../../lib/sentry';
+import { encryptAndUploadTaxFormToS3 } from '../../../lib/tax-forms';
+import { Activity, LegalDocument } from '../../../models';
+import { LEGAL_DOCUMENT_REQUEST_STATUS, LEGAL_DOCUMENT_SERVICE } from '../../../models/LegalDocument';
+import { checkRemoteUserCanUseAccount } from '../../common/scope-check';
+import { Forbidden, ValidationFailed } from '../../errors';
+import { GraphQLLegalDocumentType } from '../enum/LegalDocumentType';
+import { fetchAccountWithReference, GraphQLAccountReferenceInput } from '../input/AccountReferenceInput';
+import { GraphQLLegalDocument } from '../object/LegalDocument';
+
+const debug = debugLib('legalDocuments');
+
+export const legalDocumentsMutations = {
+  submitLegalDocument: {
+    type: new GraphQLNonNull(GraphQLLegalDocument),
+    description: 'Submit a legal document',
+    args: {
+      account: {
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'The account the legal document is for',
+      },
+      type: {
+        type: new GraphQLNonNull(GraphQLLegalDocumentType),
+        description: 'The type of legal document',
+      },
+      formData: {
+        type: new GraphQLNonNull(GraphQLJSON),
+        description:
+          'The form data for the legal document. Will be validated against the schema for the document type and encrypted.',
+      },
+    },
+    resolve: async (_, args, req) => {
+      if (!req.remoteUser) {
+        throw new Forbidden('You need to be logged in to submit a legal document');
+      }
+
+      checkRemoteUserCanUseAccount(req);
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+      if (!req.remoteUser.isAdminOfCollective(account)) {
+        throw new Forbidden('You do not have permission to submit a legal document for this account');
+      }
+
+      if (args.type === 'US_TAX_FORM') {
```

Comment:
Style: I prefer to use the pattern were we test the opposite and exit early.

```
if (args.type !== 'US_TAX_FORM') {
  throw new ValidationFailed(`Legal document type ${args.type} is not supported`);
}
```

I understand that the code as it is may be more future proof if we were to add more types.